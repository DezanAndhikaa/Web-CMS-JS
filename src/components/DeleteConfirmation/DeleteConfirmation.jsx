

import React from 'react';
import { DialogContent, Button, Modal } from '@material-ui/core';
import { Delete, DelSuccess } from '../../assets/imgs';
import './DeleteConfirmation.scss';
import CloseNotif from '../CloseNotif/CloseNotif';

export default class DeleteConfirmation extends React.PureComponent {
  render() {
    if(this.props.idDelete === "Delete"){
      return (
        <Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container-delete">
          <DialogContent className="delete-confirmation-content">
            <div className="confirmation-modal-delete">
              <CloseNotif onClose={this.props.onClose}/>
                <div className="confirmation-container-delete">
                  <p className="confirmation-title-delete">Delete Planning</p>
                  <img className="confirmation-image-delete" src={Delete} alt="" />
                  <p className="confirmation-caption-delete">Are you sure want to</p>
                  <p className="confirmation-caption-delete">delete {this.props.totalData} items?</p>
                  <div className="btn-row">
                    <Button className="btn-yes" onClick={ () => {this.props.renderSakses(); this.props.onDelete(); this.props.onClose() }}>Yes</Button> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    <Button className="btn-no" onClick={this.props.onClose}>No</Button>
                    </div>
                </div>
            </div>
          </DialogContent>
        </Modal>
      );
    } else if(this.props.idDelete === "Permanent"){
      return (
        <Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container-delete">
          <DialogContent className="delete-confirmation-content">
            <div className="confirmation-modal-delete">
              <CloseNotif onClose={this.props.onClose}/>
                <div className="confirmation-container-delete">
                  <p className="title-confirmation-delete">Are you sure want to</p>
                  <p className="confirmation-titles-delete">permanently delete this data ?</p>
                  <img className="confirmation-images-delete" src={DelSuccess} alt="" />
                  <div className="btn-row">
                    <Button className="btn-yes" onClick={this.props.onDelete}>Yes, Delete</Button> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                    <Button className="btn-no" onClick={this.props.onClose}>No</Button>
                    </div>
                </div>
            </div>
          </DialogContent>
        </Modal>
      );
    }
  }
}
