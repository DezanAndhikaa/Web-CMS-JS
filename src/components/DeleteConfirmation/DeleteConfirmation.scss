@import '../../themes/functions';
@import '../../themes/variables';

.modal-container{
    width: fit-content;
    height: fit-content;
    margin: auto;
    .confirmation-modal-content{
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 0;
        .confirmation-modal{
            width: vw(519);
            height: vh(451);
            background-color: white;
            margin: auto;
            border-radius: vw(15);
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            padding: vw(15) vw(24) vw(30) vw(24);
            .confirmation-container{
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;
                flex: 1;
                .title-confirmation{
                    margin: 0;
                    padding-top: vw(20);
                    font-size: vw(28);
                    font-family: 'SFUIDisplay-Bold';
                    color: #3C3B3B;
                }
                .confirmation-titles{
                    margin: 0;
                    font-size: vw(28);
                    padding-top: vw(10);
                    font-family: 'SFUIDisplay-Bold';
                    color: #3C3B3B;
                }
                .confirmation-images{
                    width: vw(200);
                    margin-top: vw(15);
                    margin-bottom: vw(15);
                }
                .confirmation-image{
                    width: vw(175);
                    margin-top: vw(16);
                    margin-bottom: vw(25);
                }
                .confirmation-caption{
                    margin: 0;
                    font-size: vw(20);
                    font-family: 'SFUIDisplay-Regular';
                    opacity: 0.5;
                    text-align: center;
                    color: rgb(0, 0, 0);
                }
            }
        }
    }
    .btn-row{
        flex: 1;
        display: flex;
        flex-direction: row;
        margin-top: vw(20);
        .btn-yes{
            display: flex;
            text-transform: capitalize;
            align-items: center;
            justify-content: center;
            width: 40%;//w(100.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: #F2F2F4;
            color: #1D1D1D;
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .btn-yes:hover {
            background-color: rgb(227, 227, 230);
        }
        .btn-no{
            display: flex;
            text-transform: capitalize;
            align-items: center;
            justify-content: center;
            width: 60%;//vw(170.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: map-get($colors , primary);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .btn-no:hover {
            background-color: map-get($colors, primary-dark);
        }
    }
}

@media (min-width: 600px) and (max-width: 1025px) {    
    .modal-container{
        width: fit-content;
        height: fit-content;
        margin: auto;
        .confirmation-modal-content{
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 0;
            .confirmation-modal{
                width: vw-tab(450);
                height: vw-tab(300);
                background-color: white;
                margin: auto;
                border-radius: vw-tab(5);
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;
                padding: vw-tab(15) vw-tab(24) vw-tab(30) vw-tab(24);
                .confirmation-container{
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    flex: 1;   
                    .confirmation-title{
                        margin: 0;
                        font-size: vw-tab(33);
                        font-family: 'SFUIDisplay-Bold';
                        color: #3C3B3B;
                    }
                    .confirmation-image{
                        width: vw-tab(150);
                        margin-top: vw-tab(16);
                        margin-bottom: vw-tab(25);
                    }
                    .confirmation-caption{
                        margin: 0;
                        font-size: vw-tab(20);
                        font-family: 'SFUIDisplay-Regular';
                        text-align: center;
                        opacity: 0.5;
                        color: #3C3B3B;
                    }
                }
            }
        }
    }
}

@media (max-width: 599px) {    
    .modal-container{
        width: fit-content;
        height: fit-content;
        margin: auto;
        .confirmation-modal-content{
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 0;
            .confirmation-modal{
                margin: 0;
                width: vw-mobile(215);
                height: vw-mobile(200);
                background-color: white;
                border-radius: vw-mobile(15);
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;
                padding: vw-mobile(10);
                .confirmation-container{
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    flex: 1;   
                    .confirmation-title{
                        margin: 0;
                        font-size: vw-mobile(20);
                        font-family: 'SFUIDisplay-Bold';
                        color: #3C3B3B;
                    }
                    .confirmation-image{
                        width: vw-mobile(100);
                        margin-top: vw-mobile(5);
                        margin-bottom: vw-mobile(10);
                    }
                    .confirmation-caption{
                        margin: 0;
                        font-size: vw-mobile(15);
                        font-family: 'SFUIDisplay-Regular';
                        text-align: center;
                        opacity: 0.5;
                        color: #3C3B3B;
                    }
                }
            }
        }
    }
}