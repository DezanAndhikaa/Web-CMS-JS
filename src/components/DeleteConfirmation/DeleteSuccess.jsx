

import React from 'react';
import { DialogContent, Button, Modal } from '@material-ui/core';
import { DelSuccess, Success } from '../../assets/imgs';
import './DeleteSuccess.scss';

export default class DeleteSuccess extends React.PureComponent {
  render() {
    return (
      <Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container">
        <DialogContent className="confirmation-modal-content">
          <div className="confirmation-modal">
            <div className="confirmation-container">
                <p className="confirmation-title">Successful</p>
                <img className="confirmation-image" src={Success} alt="" /> 
                <p className="confirmation-caption">The data was deleted</p>
                <p className="confirmation-caption">successfully.</p>
                <br></br>
                <Button className="btn-ok" onClick={this.props.onClose}>Continue</Button>
            </div>
          </div>
        </DialogContent>
      </Modal>
    );
  }
}
