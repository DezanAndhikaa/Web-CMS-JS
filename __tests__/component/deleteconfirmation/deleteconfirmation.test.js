import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import DeleteConfirmation from '../../../src/components/DeleteConfirmation/DeleteConfirmation';

Enzyme.configure({adapter:new Adapter()});
it("render button close notif correctly", () => {
    const wrapper = shallow (
      <DeleteConfirmation />
    )
    expect(wrapper).toMatchSnapshot();

})