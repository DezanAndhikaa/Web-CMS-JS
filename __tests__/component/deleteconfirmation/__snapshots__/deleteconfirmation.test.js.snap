exports[`render button close notif correctly 1`] = `""`;